test/controllers/sections/items/ItemPackagingProductTypeControllerSpec.scala

```diff
@@ -0,0 +1,218 @@
+          form = form,
+          packageType = packageType,
+          onSubmitAction = onSubmitAction(0, 0)
+        )(dataRequest(request, userAnswers.get), messages(request)).toString
+      }
+
+      "must populate the view correctly on a GET when the question has previously been answered" in new Test(Some(
+        emptyUserAnswers
+          .set(ItemPackagingProductTypePage(0, 0), PackagingProductType.PackagingProductTypeYes)
+          .set(ItemSelectPackagingPage(0, 0), packageType)
+      )) {
+
+        val result = controller.onPageLoad(testErn, testDraftId, 0, 0, NormalMode)(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form.fill(PackagingProductType.PackagingProductTypeYes),
+          packageType = packageType,
+          onSubmitAction = onSubmitAction(0, 0)
+        )(dataRequest(request, userAnswers.get), messages(request)).toString
+      }
+
+      "must redirect to the ItemsPackagingIndexController when no packageType has been given" in new Test(Some(emptyUserAnswers)) {
+
+        val result = controller.onPageLoad(testErn, testDraftId, 0, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual routes.ItemsPackagingIndexController.onPageLoad(testErn, testDraftId, 0).url
+      }
+
+      "must redirect to the ItemsPackagingIndexController when the packagingIndex is invalid" in new Test(Some(
+        emptyUserAnswers.set(ItemSelectPackagingPage(0, 0), packageType)
+      )) {
+
+        val result = controller.onPageLoad(testErn, testDraftId, 0, 1, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual routes.ItemsPackagingIndexController.onPageLoad(testErn, testDraftId, 0).url
+      }
+
+      "must redirect to the ItemsPackagingIndexController when the itemsIndex is invalid" in new Test(Some(
+        emptyUserAnswers.set(ItemSelectPackagingPage(0, 0), packageType)
+      )) {
+
+        val result = controller.onPageLoad(testErn, testDraftId, 1, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual routes.ItemsPackagingIndexController.onPageLoad(testErn, testDraftId, 1).url
+      }
+
+      "must redirect to Journey Recovery for a GET if no existing data is found" in new Test(None) {
+        val result = controller.onPageLoad(testErn, testDraftId, 0, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual controllers.routes.JourneyRecoveryController.onPageLoad().url
+      }
+    }
+
+    "for POST onSubmit" - {
+
+      s"must redirect to the next page when Yes is submitted" in new Test(Some(
+        emptyUserAnswers.set(ItemSelectPackagingPage(0, 0), packageType)
+      )) {
+
+        MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
```

Comment:
Is it worth passing the answers into the set call? This will ensure the correct answer is being set